python/src/spark_rapids_ml/params.py

```diff
@@ -407,16 +409,8 @@ def _set_cuml_value(self, k: str, v: Any) -> None:
             self._cuml_params[k] = v
         else:
             # value map exists
-            supported_values = set([x for x in value_map[k].values() if x is not None])
-            if v in supported_values:
-                # already a valid value
-                self._cuml_params[k] = v
+            mapped_v = value_map[k](v)
```

Comment:
If touching this common part, Seems, I guess, we need to update all the mappings in the subclass?

Reply:
I attempted to do this.   Did I miss something?

Reply:
Sorry, I interpreted this as, "all of the other subclasses need to be updated"... but upon further review, it looks like you've already updated all occurrences of `_param_value_mapping` (which does not exist for all subclasses).  So unless @wbo4958 meant something else, I think this is OK.